### PR TITLE
feature: add hooks to campaigns

### PR DESCRIPTION
**Pull request by thetutlage**
Fixes: FRMW-2645

Following is the list of hooks introduced in the promotion modules workflows.

<dl>

<dt> createCampaignsWorkflow </dt>
<dd>

- Hook: `campaignsCreated`
- Input: `{ campaigns: CampaignDTO[], additional_data?: AdditionalData }`

</dd>

<dt> updateCampaignsWorkflow </dt>
<dd>

- Hook: `campaignsUpdated`
- Input: `{ campaigns: CampaignDTO[], additional_data?: AdditionalData }`

</dd>

<dt> deleteCampaignsWorkflow </dt>
<dd>

- Hook: `campaignsDeleted`
- Input: `{ ids: string[] }`

</dd>

<dt> createPromotionsWorkflow </dt>
<dd>

- Hook: `promotionsCreated`
- Input: `{ promotions: PromotionDTO[], additional_data?: AdditionalData }`

</dd>

<dt> updatePromotionsWorkflow </dt>
<dd>

- Hook: `promotionsUpdated`
- Input: `{ promotions: PromotionDTO[], additional_data?: AdditionalData }`

</dd>

<dt> deletePromotionsWorkflow </dt>
<dd>

- Hook: `promotionsDeleted`
- Input: `{ ids: string[] }`

</dd>

</dl>

### CODE REVIEW

File: packages/medusa/src/api/admin/promotions/validators.ts [hunk 1]
```diff
@@ -155,51 +155,67 @@ const promoRefinement = (promo) => {
...
   .strict()
 
-export const AdminCreatePromotion = (customSchema?: ZodObject<any, any>) => {
-  const schema = customSchema
-    ? _AdminCreatePromotion.merge(customSchema)
-    : _AdminCreatePromotion
+export const AdminCreatePromotion = (
+  additionalDataValidator?: ZodObject<any, any>
+) => {
+  if (!additionalDataValidator) {
```
Comment (riqwan) [inline packages/medusa/src/api/admin/promotions/validators.ts:174]: q: are you doing this for all validators for every hook you create? 

Maybe worth adding a helper if its common:

```
export const AdminCreatePromotion = wrapAdditionalDataValidator(CreatePromotion)
```

**Review by riqwan — APPROVED** [review]
lgtm!

**Review by olivermrbl — APPROVED** [review]
LGTM

File: packages/medusa/src/api/admin/promotions/validators.ts (hunk 1 above)
Comment (thetutlage) [reply packages/medusa/src/api/admin/promotions/validators.ts:174]: Nice suggestion. It did felt like additional work, but I kind of tuned myself to do it. But I will create a helper now. :)

**Pull request merged**